app/lib/etl-toolkit/etlStore.js

```diff
@@ -0,0 +1,124 @@
+const log = require('../logger');
+const fsHelper = require('../fsHelper');
+const config = require('../config');
+
+const ALL_TYPE = 'all';
+
+let ids = [];
+let failedIds = {};
+let cache = {};
+
+function addRecord(record) {
+  // eslint-disable-next-line no-underscore-dangle
+  cache[record._id] = record;
```

Comment:
touched on this in the comment above, the field name could be anything or we can provide the ability to tell the etlStore which field to use as a key. _id is seemed convenient in this case.